nwb_conversion_tools/datainterfaces/behavior/movie/moviedatainterface.py

```diff
@@ -152,49 +148,54 @@ def run_conversion(
 
-            image_series_kwargs = dict(image_series_kwargs_list[j])
-            if check_regular_timestamps(ts=timestamps):
-                fps = get_movie_fps(movie_file=file)
-                image_series_kwargs.update(starting_time=starting_times[j], rate=fps)
-            else:
-                image_series_kwargs.update(timestamps=H5DataIO(timestamps, compression="gzip"))
+        image_series_kwargs_list_updated = _check_duplicates(image_series_kwargs_list)
 
+        for j, image_series_kwargs in enumerate(image_series_kwargs_list_updated):
+            file_list = image_series_kwargs.pop("data")
             if external_mode:
-                image_series_kwargs.update(format="external", external_file=[file])
+                image_series_kwargs.update(format="external", external_file=file_list)
+                with VideoCaptureContext(str(file_list[0])) as vc:
+                    fps = vc.get_movie_fps()
+                image_series_kwargs.update(starting_time=0.0, rate=fps)  # TODO manage custom starting_times
             else:
+                file = file_list[0]
                 uncompressed_estimate = Path(file).stat().st_size * 70
                 available_memory = psutil.virtual_memory().available
-                if not chunk_data and uncompressed_estimate >= available_memory:
+                if not chunk_data and not stub_test and uncompressed_estimate >= available_memory:
                     warn(
                         f"Not enough memory (estimated {round(uncompressed_estimate/1e9, 2)} GB) to load movie file as "
                         f"array ({round(available_memory/1e9, 2)} GB available)! Forcing chunk_data to True."
                     )
                     chunk_data = True
 
-                total_frames = len(timestamps)
-                frame_shape = get_frame_shape(movie_file=file)
-                maxshape = [total_frames]
-                maxshape.extend(frame_shape)
+                with VideoCaptureContext(str(file)) as video_capture_ob:
+                    total_frames = video_capture_ob.get_movie_frame_count()
+                    frame_shape = video_capture_ob.get_frame_shape()
+                    timestamps = starting_times[j] + video_capture_ob.get_movie_timestamps()
+                    fps = video_capture_ob.get_movie_fps()
+                maxshape = (total_frames, *frame_shape)
                 best_gzip_chunk = (1, frame_shape[0], frame_shape[1], 3)
                 tqdm_pos, tqdm_mininterval = (0, 10)
                 if chunk_data:
-
-                    def data_generator(file, count_max):
-                        cap = cv2.VideoCapture(str(file))
-                        for _ in range(min(count_max, total_frames)):
-                            success, frame = cap.read()
-                            yield frame
-                        cap.release()
-
-                    mov = DataChunkIterator(
+                    _ = video_capture_ob.__enter__()
```

Comment:
This is very highly irregular usage of the magic function. Is it not able to be opened and used as an iterable like in https://github.com/catalystneuro/nwb-conversion-tools/blob/main/tests/test_internals/test_movie_utils.py#L71-L78?

Reply:
this just reinstantiates the `self.vc=cv2.VideoCapture()`, but no the whole object. We could however reinstantiate since its not expensive

Reply:
Yeah, I would always re-instantiate explicitly; the `__enter__` method really should never be called explicitly like that